src/api/HttpApi.ts

```diff
@@ -172,27 +183,32 @@ export default class HttpApi {
     return `${pathBase}?${search}`;
   }
 
-  createArgLoader(path: string, key: string) {
-    return this.createLoader(
+  createArgLoader<T extends Record<string, unknown>>(
+    path: string,
+    key: string,
+  ) {
+    return this.createLoader<T>(
       keys => this.getUrl(path, { [key]: keys }),
-      item => item[key],
+      item => item[key] as string,
     );
   }
 
-  createLoader<T>(
+  createLoader<T extends Record<string, unknown>>(
     getPath: (keys: string[]) => string,
     getKey: (obj: T) => string,
   ) {
-    return new DataLoader<*, *>(async keys => {
+    return new DataLoader<any, any>(async keys => {
```

Comment:
`any` is different than `*` here, since * means "infer", I'm not sure if we were getting any safety out of it, but this is theoretically less safe.